tests/integration/src/test/scala/mesosphere/marathon/integration/AgentGoneByOperatorIntegrationTest.scala

```diff
@@ -0,0 +1,59 @@
+package mesosphere.marathon
+package integration
+
+import mesosphere.AkkaIntegrationTest
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.integration.facades.MarathonFacade._
+import mesosphere.marathon.integration.setup.{EmbeddedMarathonTest, RestResult}
+import mesosphere.marathon.raml.App
+import mesosphere.marathon.state.PathId
+import org.scalatest.Inside
+
+class AgentGoneByOperatorIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest with Inside {
+  override lazy val mesosNumSlaves = 2
+
+  "resident task will be expunged and a new instance will be created in response to TASK_GONE_BY_OPERATOR" in {
+    Given("An app with a persistent volume")
+    val containerPath = "persistent-volume"
+    val id = appId("resident-task-with-persistent-volume-will-expunge-with-gone-by-operator")
+    val app = residentApp(
+      id = id,
+      containerPath = containerPath,
+      cmd = "sleep 36000")
+
+    When("a task is launched")
+    createSuccessfully(app)
+
+    And("the matching agent is marked gone")
+    val task = inside(marathon.tasks(id).value) {
```

Comment:
`val task = marathon.tasks(id).value.head` would work as well, wouldn't it? Or `val Seq(task) = marathon.tasks(id).value` as you do below.

Reply:
Yeah, those would work. What's wrong with inside?